src/app/pheno-browser-table/pheno-browser-table.component.ts

```diff
@@ -1,58 +1,119 @@
+  OnInit,
+  Component,
+  HostListener,
+  Input,
+  ViewChildren,
+  ChangeDetectionStrategy
 } from '@angular/core';
-
 import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
 import {
   PhenoBrowserModalContentComponent
 } from '../pheno-browser-modal-content/pheno-browser-modal-content.component';
-import { PhenoMeasures } from '../pheno-browser/pheno-browser';
+import { PhenoMeasure, PhenoMeasures } from '../pheno-browser/pheno-browser';
+import { SortingButtonsComponent } from 'app/sorting-buttons/sorting-buttons.component';
+import { isNumber, isString } from 'lodash';
 
 @Component({
   selector: 'gpf-pheno-browser-table',
   templateUrl: './pheno-browser-table.component.html',
-  styleUrls: ['./pheno-browser-table.component.css']
+  styleUrls: ['./pheno-browser-table.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class PhenoBrowserTableComponent implements OnInit, OnChanges {
+export class PhenoBrowserTableComponent implements OnInit {
   @Input() public measures: PhenoMeasures;
-  @ViewChild('templateRef', { read: TemplateRef }) private templateRef: TemplateRef<Element>;
-  @ViewChild('viewContainerRef', { read: ViewContainerRef }) private viewContainerRef: ViewContainerRef;
+  @Input() public measuresChangeTick: number;
+  @ViewChildren(SortingButtonsComponent) public sortingButtonsComponents: SortingButtonsComponent[];
+
 
   public singleColumnWidth;
-  private columnsCount = 8;
+  public columnsCount = 4;
+  public sortBy: string;
+  public orderBy = 'desc';
 
   public constructor(
     private modalService: NgbModal
   ) { }
 
   public ngOnInit(): void {
+    this.columnsCount += 2*Object.keys(this.measures.regressionNames).length;
     if (this.measures.hasDescriptions) {
-      this.columnsCount = 9;
+      this.columnsCount += 1;
     }
     this.onResize();
   }
 
-  public ngOnChanges(): void {
-    this.viewContainerRef?.clear();
-    this.viewContainerRef?.createEmbeddedView(this.templateRef);
+  public sort(sortBy: string, orderBy: string): void {
+    if (this.sortBy !== sortBy) {
+      this.resetSortButtons();
+    }
+
+    this.sortBy = sortBy;
+    this.orderBy = orderBy;
+
+    const sortButton = this.sortingButtonsComponents.find(
+      sortingButtonsComponent => sortingButtonsComponent.id === this.sortBy
+    );
+
+    if (sortButton) {
+      sortButton.emitSort();
+    }
+
+    this.sortTable(this.measures.measures, this.orderBy, this.sortBy);
+  }
+
+  public sortTable(data: PhenoMeasure[], orderBy: string, sortBy: string): void {
+    data.sort((m1, m2) => {
+      let a: string | number;
+      let b: string | number;
+      if (sortBy.includes('pvalueRegressionMale')) {
+        const arr = sortBy.split('.');
+        a = m1.regressions.getReg(arr[0]).pvalueRegressionMale;
+        b = m2.regressions.getReg(arr[0]).pvalueRegressionMale;
+      } else if (sortBy.includes('pvalueRegressionFemale')) {
+        const arr = sortBy.split('.');
```

Comment:
Change const arr = sortBy.split('.'); to const regId = sortBy.split('.')[0];
And down use regId